services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -597,6 +598,58 @@ class ApplicationServiceITest : DatabaseTest() {
             }
         }
 
+        @Test
+        fun `when application is in Allu should initialize access for new contacts`() {
```

Comment:
Since we recently realized we're sending several emails to the same person if they're added to different roles in the same application, there should be a test that documents this behaviour. Even if we don't make any changes to the behaviour yet.

Also, does this one test cover all possible cases?

Reply:
Can you elaborate what do you mean by all possible cases? If there is an email that did not exist on the previous application state -> send email. The subtractByEmail function could be more thoroughly tested, but maybe not in an integration test?

Reply:
I'm not sure which cases should be tested. I'm just worried that there's just one test added.

Maybe the case where the current user doesn't have a HankeKayttaja?

Or different customer/contact combinations:
- New customer / old customer
- New contact under an existing customer
- Several contacts under the same customer

Is there anything outside the happy path that should be tested?

Of course, all the tests don't have to be top-level integration tests.